zkabacus-crypto/src/revlock.rs

```diff
@@ -168,52 +216,78 @@ impl RevocationLockCommitment {
     pub(crate) fn verify_revocation_pair(
         &self,
         parameters: &merchant::Config,
-        revocation_secret: &RevocationSecret,
-        revocation_lock: &RevocationLock,
+        revocation_pair: &RevocationPair,
         revocation_lock_blinding_factor: &RevocationLockBlindingFactor,
     ) -> Verification {
-        let pair_is_valid = revocation_lock.verify(revocation_secret);
         let opening_is_valid = self.0.verify_opening(
             parameters.revocation_commitment_parameters(),
             revocation_lock_blinding_factor.0,
-            &Message::from(revocation_lock.to_scalar()),
+            &Message::from(revocation_pair.lock.to_scalar()),
         );
 
-        match (pair_is_valid, opening_is_valid) {
-            (Verification::Verified, true) => Verification::Verified,
-            _ => Verification::Failed,
+        if opening_is_valid {
+            Verification::Verified
+        } else {
+            Verification::Failed
         }
     }
 }
 
 #[cfg(test)]
 mod test {
     use super::*;
+    use std::convert::TryInto;
     use {hex, rand::thread_rng};
 
     #[test]
-    pub fn revlock_is_correct() {
-        let rs = RevocationSecret::new(&mut thread_rng());
-        let rl = rs.revocation_lock();
-        assert!(matches!(rl.verify(&rs), Verification::Verified));
+    // Test computation of a verified revocation pair from
+    // - a valid secret
+    // - a valid pair
+    pub fn revpair_is_correct() {
+        let mut rng = thread_rng();
+
+        for _ in 1..1000 {
+            let rp = RevocationPair::new(&mut rng);
+
+            let maybe_rs = UncheckedRevocationSecret {
+                secret: rp.secret.secret,
+                index: rp.secret.index,
+            };
+
+            let maybe_rp_from_unchecked_pair = RevocationPair::try_from(UncheckedRevocationPair {
+                secret: maybe_rs,
+                lock: rp.lock,
+            })
+            .unwrap();
+            let maybe_rp_from_secret = RevocationPair::try_from(maybe_rs).unwrap();
```

Comment:
Nit on naming: I would not call these three things `maybe_`s, because you're actively unwrapping them. There is no situation where they are maybe `Some` and maybe `None`: they're either a `RevocationPair` or the program panics.

The behavior is fine, but I would remove the `maybe_`s from the names.